SwitchSpeak-iOS/SwitchSpeak-iOS/TouchGrid/Tree/Node.swift

```diff
@@ -47,7 +47,31 @@ class Node {
 		for childNode in self.childNodes {
 			childNode.unHighlightSubTree()
 		}
-	}	
+	}
+    
+    // it will retrun a node. if it has only undummy child node, it will return that child node, else it will return itself
```

Comment:
Please change this comment to:
`// If this node only has one non-dummy child node, that node is returned by this function; otherwise the function returns the node itself`
This says what you were saying, but it's a little easier to read. Clarity is key here!